test/e2e/prometheus_test.go

```diff
@@ -5495,6 +5495,79 @@ func testPrometheusReconciliationOnSecretChanges(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func testPrometheusUTF8LabelSupport(t *testing.T) {
+	skipPrometheusTests(t)
+	t.Parallel()
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	name := "prometheus-utf8-test"
+
+	// Create service with UTF8 labels
```

Comment:
see https://github.com/prometheus-operator/prometheus-operator/pull/7955. After it merges, I can push an updated version of instrumented-app which exposes a metric with dots so we can test with a "real" example.